tasty-query/shared/src/main/scala/tastyquery/Printers.scala

```diff
@@ -204,6 +192,545 @@ private[tastyquery] object Printers:
+        print("(")
+        print(selector)
+        print(" match")
+        if !isSyntacticAny(bound.toType) then
+          print(" <: ")
+          print(bound)
+        printBlock(cases)(print(_))
+        print(")")
+
+      case TypeTreeBind(name, body, symbol) =>
+        print(name)
+
+      case TypeLambdaTree(tparams, body) =>
+        print("([")
+        printCommaSeparatedList(tparams)(print(_))
+        print("] =>> ")
+        print(body)
+        print(")")
+
+      case TypeBindingsTree(bindings, body) =>
+        printBlock[TypeMember | TypeTree](bindings, body) { elem =>
+          elem match
+            case elem: TypeMember => print(elem)
+            case elem: TypeTree   => print(elem)
+        }
+    end print
+
+    def print(tree: WildcardTypeArgTree): Unit =
+      print("?")
+      print(tree.bounds)
+    end print
+
+    def print(tree: TypeCaseDef): Unit =
+      print("case ")
+      print(tree.pattern)
+      print(" => ")
+      print(tree.body)
+    end print
+
+    def print(tree: TypeDefinitionTree): Unit = tree match
+      case InferredTypeBoundsTree(bounds) =>
+        print(bounds)
+
+      case ExplicitTypeBoundsTree(low, high) =>
+        if !isSyntacticNothing(low.toType) then
+          print(" >: ")
+          print(low)
+        if !isSyntacticAny(high.toType) then
+          print(" <: ")
+          print(high)
+
+      case TypeAliasDefinitionTree(alias) =>
+        alias match
+          case MatchTypeTree(bound, selector, cases) =>
+            if !isSyntacticAny(bound.toType) then
+              print(" <: ")
+              print(bound)
+            print(" = ")
+            print(selector)
+            print(" match")
+            printBlock(cases) { caze =>
+              print("case ")
+              print(caze.pattern)
+              print(" => ")
+              print(caze.body)
+            }
+          case _ =>
+            print(" = ")
+            print(alias)
+
+      case OpaqueTypeAliasDefinitionTree(bounds, alias) =>
+        print(bounds)
+        print(alias)
+
+      case PolyTypeDefinitionTree(tparams, body) =>
+        print("[")
+        printCommaSeparatedList(tparams)(print(_))
+        print("]")
+        print(body)
+
+      case NamedTypeBoundsTree(name, bounds) =>
+        print(name)
+    end print
+
+    final def printBlock[A](initElems: List[A], lastElem: A)(printElem: A => Unit): Unit =
+      printBlock(initElems :+ lastElem)(printElem)
+
+    def printCommaSeparatedList[A](elems: List[A])(printElem: A => Unit): Unit =
+      if elems.nonEmpty then
+        printElem(elems.head)
+        for elem <- elems.tail do
+          print(", ")
+          printElem(elem)
+    end printCommaSeparatedList
+
+    def printBlock[A](elems: List[A])(printElem: A => Unit): Unit =
+      if elems.isEmpty then print(" {}")
+      else
+        print(" { ")
+        printElem(elems.head)
+        for elem <- elems.tail do
+          print("; ")
```

Comment:
Would it be more convenient to have a line break here?

```suggestion
          print("\n")
```

Reply:
I guess then there should be a stack of indentation levels to respect

Reply:
Yes, if we put newlines we must deal with indentation. My thought was that it would be left for a subclass that can surgically override `printBlock`.

Reply:
I don't think we need any indentation since we have "{ }" around all blocks.